src/ui/form-modal-ui-handler.ts

```diff
@@ -45,37 +50,50 @@ export abstract class FormModalUiHandler extends ModalUiHandler {
   setup(): void {
     super.setup();
 
-    const fields = this.getFields();
+    const config = this.getInputFieldConfigs();
 
     const hasTitle = !!this.getModalTitle();
 
-    fields.forEach((field, f) => {
-      const label = addTextObject(this.scene, 10, (hasTitle ? 31 : 5) + 20 * f, field, TextStyle.TOOLTIP_CONTENT);
+    if (config.length >= 1) {
+      this.updateFields(config, hasTitle);
+    }
+
+    this.errorMessage = addTextObject(this.scene, 10, (hasTitle ? 31 : 5) + 20 * (config.length - 1) + 16 + this.getButtonTopMargin(), "", TextStyle.TOOLTIP_CONTENT);
+    this.errorMessage.setColor(this.getTextColor(TextStyle.SUMMARY_PINK));
+    this.errorMessage.setShadowColor(this.getTextColor(TextStyle.SUMMARY_PINK, true));
+    this.errorMessage.setVisible(false);
+    this.modalContainer.add(this.errorMessage);
+  }
+
+  updateFields(fieldsConfig: InputFieldConfig[], hasTitle: boolean) {
```

Comment:
can we get tsdoc for this? can probably be `protected` too

```suggestion
  protected updateFields(fieldsConfig: InputFieldConfig[], hasTitle: boolean) {
```